allennlp/modules/default_lstm.py

```diff
@@ -0,0 +1,36 @@
+import re
+from overrides import overrides
+import torch
+
+from allennlp.nn.initializers import block_orthogonal
+
+
+class DefaultLstm(torch.nn.LSTM):
+    """
+    A wrapper of the Pytorch LSTM which provides a good default initialization
+    of weights and biases, and disables training of the superfluous bias_hh parameters.
+
+    Input parameters are identical to Pytorch LSTM.
+    """
+    @overrides
+    def reset_parameters(self):
+        hidden_size = self.hidden_size
+        for param_name, param in self.named_parameters():
+            if re.search("bias_hh", param_name):
+                torch.nn.init.constant(param, 0)
+                # This parameter is superfluous, so we should not update it in training
+                param.requires_grad = False
```

Comment:
I don't think this is worth the confusion for an initial user of having to do the `parameters = [p for p in self.model.parameters() if p.requires_grad]` trick were they to try to use this LSTM with their own training code. It's also pretty harmless.

Reply:
Right, that might be confusing, I don't have a good sense of that. I'm not sure how harmless or not it is. It seems unnecessary to me to have hundreds of completely flat directions in your parameter space, although it's a small enough fraction of the total parameters that it practically might not matter much. Maybe I can make it a boolean flag which is off by default to avoid this annoyance with parameters (note we do this `requires_grad` filtering by default in our training code since we also have things like word embeddings which maybe be fixed, so it's probably general good practice to do this). We could also add a method to our `Model` class to get trainable parameters which does this filtering for you.